### ISSUE TITLE
last.fm intigration

### ISSUE THREAD

**Issue by simonv3** [KeitIG/museeks#231]

**Comment by KeitIG** [comment]
Hello, I've never used this service could you explain me what last.fm does exactly ?

**Comment by simonv3** [comment]
Oh, sorry!

So, this is interesting - Last.fm is technically a radio music player (a bit like Pandora), but as far as I'm aware most people I know who use it use it as a what-they've-listened-to tracker. For example, I can look at what artists I listened to the most in 2007. It's a very inactive process - Spotify lets you add your last.fm account and it just automatically tells last.fm what you've been listening to. As a user you don't have to do anything else. 

It would be a shame (for me) to lose that tracked information with switching back to a more offline, I-own-the-music, open-source service. 

I'm personally willing to look into getting this implemented for Museeks, but just wanted to see if it was a starter or not.

**Comment by KeitIG** [comment]
Thanks for the explanation, looks like an interesting feature indeed.

Low priority for the moment, but it may go up in the backlog if other people ask for it :) Tanks for the suggestion.

**Comment by YurySolovyov** [comment]
I will actually be against this, not because it is not useful, but because it adds [feature creep](https://en.wikipedia.org/wiki/Feature_creep).
For me, everything that is not playback-related, is unnecessary.

**Comment by simonv3** [comment]
@YurySolovyov - that's a fair concern! 

It makes me wonder - have you all considered a light-weight plug in system? That way you can keep Museeks trim, but allow people to modify it with plug-ins? That might also be feature creep.

**Comment by YurySolovyov** [comment]
Plugin systems are usually very hard to get right.

I thought about adding a statement that we are not responsible for any issues caused by plugins, but this makes me think that it is better not to have a plugin system than having one we don't care about.

Another point is that we did not released 1.0 version yet, so things would change a lot and I'm not sure plugin system is on the roadmap.

**Comment by KeitIG** [comment]
Adding the discussion label.

**Comment by vvesperr** [comment]
Museeks has a beautiful UI. Sad to see no LastFM integration yet. It's a bit of a dealbreaker for me :( (and some others who've been using the service for a long time too, I imagine.)

**Comment by KeitIG** [comment]
@vvesperr I'm not decided yet, but thanks for your feedback, be sure it's taken in consideration :)

**Comment by meradoou** [comment]
LastFM or LibreFM scrobble

**Comment by balzabalza** [comment]
I do agree with this possible feature. With the latest releases, Museeks has made very good progresses, and I use this feature a lot, to keep track of what I'm listening to via PC/Smartphone/Tablet, no matter the program that is responsible for playing music (iTunes/Museeks/Shuttle+/Spotify...).

Features like this could be a deal breaker for people like me, although of course it is not of the highest priority.

**Comment by KeitIG** [comment]
Well, it seems to gain traction :) I may re-prioritize this then

**Comment by felixding** [comment]
Can you please add last.fm integration to Museeks? This really is a deal breaker for a long-time last.fm user like me.

**Comment by simonv3** [comment]
@martpie that's right, just the scrobbling. I dunno, it's a self-tracking thing I guess.

**Comment by martpie** [comment]
Ok, with the latest refactoring on the main process, it should not be too hard to implement in a nice and clean way. This received enough +1 so I am willing to implement it :) 

Let's code!

**Comment by balzabalza** [comment]
Thank you :)

**Comment by felixding** [comment]
Yes. Just the scrobbling.

Thank you. Looking forward to the new version.

Fingers crossed.

**Comment by meradoou** [comment]
thank you

**Comment by ldexterldesign** [comment]
👋 https://github.com/martpie/museeks/issues/518

**Comment by xE0F9** [comment]
still no support :(

**Comment by martpie** [comment]
I gave it a try last year really briefly (I worked a couple of hours on it).

The main problem is their authentication system is extremely annoying to deal with. You have to redirect to last.fm, then it will redirect to your app (if it's even possible), etc.

If it had a token system like GitHub, things would be much easier.

I don't have the use of last.fm, but if anyone knows an open-source player that does this, with some code I could look at, I could give it another try.

**Comment by ldexterldesign** [comment]
Hi @xE0F9,

Did you try [ListenBrainz]?

Sincerely

[ListenBrainz]: https://github.com/martpie/museeks/issues/518

**Comment by martpie** [comment]
ListenBrainz (after a really brief look), looks easier to implement (again, just for basic scrobbling)

**Comment by ldexterldesign** [comment]
@martpie have had to go back to iTunes until https://github.com/martpie/museeks/issues/519 and https://github.com/martpie/museeks/issues/528 get resolved but if/when that happens I'll commit to doing https://github.com/martpie/museeks/issues/518 - fair?

Sincerely

**Comment by xE0F9** [comment]
If ListenBrainz gets implemented then I would use that.

**Comment by undergroundwires** [comment]
> I don't know Last.fm much, do we agree that you all ask for Last.fm's scrobbling integration only? (and not the all the other fancy stuff like playlists etc...?)

And also Love button would great. Red when a song is loved and empty (clickable) when it's not.

**Comment by meeniemarv** [comment]
Libre.fm scrobbling would be also very welcome. It's the open source fediverse version of last.fm

https://github.com/libre-fm/developer

**Comment by martpie** [comment]
Soooo, I just asked Claude to do it and it did a pretty good job, scrobble seems to be working, I'm not so sure about the now-playing part. 

Again, I'm very noob to scrobblers, so I'm discovering that by implementing it. I still have some security/distribution concerns.

It's not really ready for prime time yet, I would like to see how to make it work with other scrobblers like Musicbrainz, @meeniemarv  just mentioned libre-fm, etc.
